test/ExecuteUpdateTest.cpp

```diff
@@ -32,6 +32,165 @@ MATCHER_P(AlwaysFalse, msg, "") {
+          "Only INSERT/DELETE update operations are currently supported."));
+}
+
+// _____________________________________________________________________________
+TEST(ExecuteUpdate, computeGraphUpdateQuads) {
+  // These tests run on the default dataset defined in
+  // `IndexTestHelpers::makeTestIndex`.
+  QueryExecutionContext* qec = ad_utility::testing::getQec(std::nullopt);
+  const Index& index = qec->getIndex();
+  const auto Id = ad_utility::testing::makeGetId(index);
+  auto defaultGraphId = Id(std::string{DEFAULT_GRAPH_IRI});
+
+  ad_utility::HashMap<std::string, std::unique_ptr<LocalVocabEntry>>
+      localVocabEntries;
+  auto LVI = [&localVocabEntries](const std::string& iri) {
+    if (!localVocabEntries.contains(iri)) {
+      localVocabEntries.try_emplace(
+          iri, std::make_unique<LocalVocabEntry>(
+                   ad_utility::triple_component::Iri::fromIriref(iri)));
+    }
+    return Id::makeFromLocalVocabIndex(localVocabEntries.at(iri).get());
+  };
+
+  auto IdTriple = [defaultGraphId](const ::Id s, const ::Id p, const ::Id o,
+                                   const std::optional<::Id> graph =
+                                       std::nullopt) -> ::IdTriple<> {
+    return ::IdTriple({s, p, o, graph.value_or(defaultGraphId)});
+  };
+
+  auto executeComputeGraphUpdateQuads = [&qec,
+                                         &index](const std::string& update) {
+    const auto sharedHandle =
+        std::make_shared<ad_utility::CancellationHandle<>>();
+    const std::vector<DatasetClause> datasets = {};
+    auto pq = SparqlParser::parseQuery(update);
+    QueryPlanner qp{qec, sharedHandle};
+    const auto qet = qp.createExecutionTree(pq);
+    return ExecuteUpdate::computeGraphUpdateQuads(index, pq, qet, sharedHandle);
+  };
+  auto expectComputeGraphUpdateQuads =
+      [&executeComputeGraphUpdateQuads](
+          const std::string& update,
+          const testing::Matcher<const std::vector<::IdTriple<>>&>&
+              toInsertMatcher,
+          const testing::Matcher<const std::vector<::IdTriple<>>&>&
+              toDeleteMatcher) {
+        EXPECT_THAT(
+            executeComputeGraphUpdateQuads(update),
+            testing::Pair(AD_FIELD(ExecuteUpdate::IdTriplesAndLocalVocab,
+                                   idTriples, toInsertMatcher),
+                          AD_FIELD(ExecuteUpdate::IdTriplesAndLocalVocab,
+                                   idTriples, toDeleteMatcher)));
+      };
+  auto expectComputeGraphUpdateQuadsFails =
+      [&executeComputeGraphUpdateQuads](
+          const std::string& update,
+          const testing::Matcher<const std::string&>& messageMatcher) {
+        AD_EXPECT_THROW_WITH_MESSAGE(executeComputeGraphUpdateQuads(update),
+                                     messageMatcher);
+      };
+
+  expectComputeGraphUpdateQuads(
+      "INSERT DATA { <s> <p> <o> . }",
+      testing::ElementsAreArray({IdTriple(LVI("<s>"), LVI("<p>"), LVI("<o>"))}),
```

Comment:
I think inside tests a `using namespace ::testing` makes stuff shorter (There is only one `ElementsAreArray` and `IsEmpty`.